padd.sh

```diff
@@ -1011,6 +1011,28 @@ VersionConverter() {
   echo "$@" | tr -d '[:alpha:]' | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }';
 }
 
+# get the Telnet API Port FTL is using by parsing `pihole-FTL.conf`
+# same implementation as https://github.com/pi-hole/pi-hole/pull/4945
+getFTLAPIPort(){
+    local FTLCONFFILE="/etc/pihole/pihole-FTL.conf"
+    local DEFAULT_FTL_PORT=4711
+    local ftl_api_port
+
+    if [ -s "$FTLCONFFILE" ]; then
+        # if FTLPORT is not set in pihole-FTL.conf, use the default port
+        ftl_api_port="$({ grep '^FTLPORT=' "${FTLCONFFILE}" || echo "${DEFAULT_FTL_PORT}"; } | cut -d'=' -f2-)"
```

Comment:
Gut feeling says there is a better way than piping `echo` to `cut` but that may involve `sed`/`awk` and that's not something to wish upon a friend!

Reply:
We use the same code in a few places in `core`. The `echo` pipe could be circumvented with something like

```
ftl_api_port="$(grep '^FTLPORT=' "${FTLCONFFILE}" | cut -d'=' -f2-)"

if [ -z  "${ftl_api_port}" ]; then
  ftl_api_port="${DEFAULT_FTL_PORT}"
fi
```